scour/scour.py

```diff
@@ -1800,9 +1800,21 @@ def repairStyle(node, options):
                 del styleMap['overflow']
                 num += 1
 
-        # now if any of the properties match known SVG attributes we prefer attributes
-        # over style so emit them and remove them from the style map
-        if options.style_to_xml:
+    if node.nodeType == Node.ELEMENT_NODE:
+        if options.style_type == "inline":
+            # Prefer inline style
+            # Remove known SVG attributes and store their values in style attribute
+            attributes = [node.attributes.item(i).nodeName for i in range(node.attributes.length)]
+            for attribute in attributes:
+                if attribute in svgAttributes:
+                    styleMap[attribute] = node.getAttribute(attribute)
+                    node.removeAttribute(attribute)
```

Comment:
Drive by-remark, I think the following will be slightly better as it avoids another attribute lookup.

```suggestion
            attributes = [node.attributes.item(i) for i in range(node.attributes.length)]
            for attribute in attributes:
                name = attribute.nodeName
                if name in svgAttributes:
                    styleMap[name] = attribute.nodeValue
                    node.removeAttribute(name)
```

(note: hand written directly in github, it may contain typoes).

Otherwise, this part LGTM.